Polygon_mesh_processing/include/CGAL/Polygon_mesh_processing/non_rigid_mesh_registration.h

```diff
@@ -0,0 +1,930 @@
+* @tparam TriangleMesh2 a const model of the `MutableFaceGraph`.
+* @tparam VertexTranslationMap is a property map with `boost::graph_traits<TriangleMesh1>::%vertex_descriptor`
+*   as key type and a \cgal Kernel `Vector_3` as value type.
+* @tparam VertexRotationMap is a property map with `boost::graph_traits<TriangleMesh1>::%vertex_descriptor`
+*   as key type and a \cgal Kernel `Aff_transformation_3` as value type.
+* @tparam NamedParameters1 a sequence of \ref bgl_namedparameters "Named Parameters1".
+* @tparam NamedParameters2 a sequence of \ref bgl_namedparameters "Named Parameters2".
+*
+* @param source the triangle mesh to be mapped onto `target`.
+* @param target the target triangle mesh.
+* @param vtm a writable vertex property map of `source` to store the translation vector of the registration.
+* @param vrm a writable vertex property map of `source` to store the rotation part of the registration.
+* @param np1 an optional sequence of \ref bgl_namedparameters "Named Parameters 1" of the `source` and the method among the ones listed below.
+* @param np2 an optional sequence of \ref bgl_namedparameters "Named Parameters 2" of the `target` providing a vertex point map and a vertex normal map as listed below.
+*
+* \cgalNamedParamsBegin{Named Parameters 1}
+*   \cgalParamNBegin{number_of_iterations}
+*     \cgalParamDescription{the number of registration iterations using ICP, iterative closest point}
+*     \cgalParamType{unsigned int}
+*     \cgalParamDefault{`50`}
+*   \cgalParamNEnd
+*
+*   \cgalParamNBegin{point_to_plane_energy}
+*     \cgalParamDescription{the weight of the point to plane distance in the registration}
+*     \cgalParamType{double}
+*     \cgalParamDefault{`1`}
+*   \cgalParamNEnd
+*
+*   \cgalParamNBegin{point_to_point_energy}
+*     \cgalParamDescription{the weight of the point to matching point distance in the registration}
+*     \cgalParamType{double}
+*     \cgalParamDefault{`1`}
+*   \cgalParamNEnd
+*
+*   \cgalParamNBegin{as_rigid_as_possible_energy}
+*     \cgalParamDescription{defines the rigidity of the registration}
+*     \cgalParamType{double}
+*     \cgalParamDefault{`50`}
+*   \cgalParamNEnd
+*
+*   \cgalParamNBegin{maximum_matching_distance}
+*     \cgalParamDescription{the maximum distance for a vertex in `source` to match with a point in `target`. The default value 0 means no maximum matching distance.}
+*     \cgalParamType{double}
+*     \cgalParamDefault{`0`}
+*   \cgalParamNEnd
+*
+*   \cgalParamNBegin{correspondences}
+*     \cgalParamDescription{a range of matching vertex pairs between the `source` and the `target`.}
+*     \cgalParamType{`ConstRange` whose value type is a pair of `boost::graph_traits<TriangleMesh1>::%vertex_descriptor` and `boost::graph_traits<TriangleMesh2>::%vertex_descriptor`.}
+*     \cgalParamDefault{empty}
+*     \cgalParamExtra{to avoid copies, this parameter can be passed using `std::cref`.}
+*   \cgalParamNEnd
+*
+*   \cgalParamNBegin{vertex_point_map}
```

Comment:
no vector map for TM1?

Reply:
Normals of the source mesh are not used in the registration. The normals of the target mesh/point cloud are used for the point to plane energy.